alica_engine/include/engine/RunnableObject.h

```diff
@@ -81,10 +82,7 @@ class RunnableObject
     virtual void doRun() = 0;
     virtual void doTerminate() = 0;
 
-    void setTracing(TracingType type, std::function<std::optional<std::string>()> customTraceContextGetter = {})
-    {
-        _runnableObjectTracer.setTracing(type, customTraceContextGetter);
-    }
+    void setTracing(TracingType type, std::function<std::optional<std::string>()> customTraceContextGetter = {});
```

Comment:
this is a bit complicated. Instead, can set the logger in the constructor & store it in the TraceRunnableObject as a protected member. We want to remove engine references in all modules as well